### PR TITLE
Sputnik DAOs routing, page, styling

### PR DESCRIPTION
**Pull request by true-eye**
image

### CODE REVIEW

File: client/scripts/views/pages/sputnikdaos.ts [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+  proposalBond: string;
+  proposalPeriod: string;
+  bountyBond: string;
+  bountyPeriod: string;
+  council: string[];
+}
+
+const SputnikDAOsPage : m.Component<{}, { daosRequested: boolean, daosList: IDaoInfo[] }> = {
+  view: (vnode) => {
+    const activeEntity = app.community ? app.community : app.chain;
```
Comment (zakhap) [inline client/scripts/views/pages/sputnikdaos.ts:30]: check here if `near` or not. if not, redirect to `/:scope`

**Review by zakhap — CHANGES_REQUESTED** [review]
We're also gonna need a solution for when you click on a link to a dao community and it doesn't yet exist (currently it 404s). Either we only link when the community already exists (filtering those communities to the top of the DAO list page) or...

File: client/scripts/views/pages/sputnikdaos.ts [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+      m(Table, [
+        m('tr', [
+          m('th', {style: {width: "27%"}}, 'Name'),
+          m('th', {style: {width: "20%"}}, 'Dao Funds ', [ m('span.nearBadge', 'NEAR') ]),
+          m('th', {style: {width: "17%"}}, 'Council'),
+          m('th', {style: {width: "19%"}}, 'Bond ', [ m('span.nearBadge', 'NEAR') ]),
+          m('th', {style: {width: "17%"}}, 'Vote Period'),
+        ]),
+        vnode.state.daosList.map((dao) => {
+          return m('tr.nearRow', {
```
Comment (zakhap) [inline client/scripts/views/pages/sputnikdaos.ts:85]: actually let's factor these rows out into it's own component, with a property to turn onclick redirect on/off. We'll also only turn on the redirect linking if the community already exists. 

also, we should sort the communities first if they exist on cw, then by bond.

File: client/scripts/views/pages/sputnikdaos.ts [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+        m(CommunityOptionsPopover, { isAdmin, isMod }),
+        m(Tag, { size: 'xs', label: 'Beta', style: 'position: relative; top: -2px; margin-left: 6px' })
+      ],
+      showNewProposalButton: true,
+    }, [
+      m('.title', 'Sputnik DAOs'),
+      m(Table, [
+        m('tr', [
+          m('th', {style: {width: "27%"}}, 'Name'),
+          m('th', {style: {width: "20%"}}, 'Dao Funds ', [ m('span.nearBadge', 'NEAR') ]),
```
Comment (zakhap) [inline client/scripts/views/pages/sputnikdaos.ts:79]: as for these big numbers, we should parse them like in `sputnik/proposal.ts`:

```
  // percentage of voters or percentage of tokens depending on type of vote
  get turnout() {
    const PRECISION = 10_000;
    const [yes, no, remove] = this._getVoteCounts();
    const totalVoted = yes.add(no).add(remove);
    see >>>> const pctVoted = totalVoted.muln(PRECISION).div(this._totalSupply).toNumber() / PRECISION;
    return pctVoted;

```

File: client/scripts/views/pages/sputnikdaos.ts [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+        ]),
+        vnode.state.daosList.map((dao) => {
+          return m('tr.nearRow', {
+            onclick: (e) => {
+              e.preventDefault();
+              m.route.set(`/${dao.contractId}`);
+            }
+          }, [
+            m('td', dao.name),
+            m('td', dao.amount),
```
Comment (zakhap) [inline client/scripts/views/pages/sputnikdaos.ts:92]: as for these big numbers, we should parse them like in `sputnik/proposal.ts`:

```
  // percentage of voters or percentage of tokens depending on type of vote
  get turnout() {
    const PRECISION = 10_000;
    const [yes, no, remove] = this._getVoteCounts();
    const totalVoted = yes.add(no).add(remove);
    see >>>> const pctVoted = totalVoted.muln(PRECISION).div(this._totalSupply).toNumber() / PRECISION;
    return pctVoted;

```

File: client/scripts/views/pages/sputnikdaos.ts (hunk 1 above)
Comment (true-eye) [reply client/scripts/views/pages/sputnikdaos.ts:30]: OK, Got it.

File: client/styles/pages/sputnikdaos.scss [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+    .nearBadge{
+        display: inline-block;
+        height: 20px;
+        background: #EAEAEA;
+        border-radius: 4px;
+        font-size: 14px;
+        padding: 3px 3px;
+    }
+
+    th{
```
Comment (zakhap) [inline client/styles/pages/sputnikdaos.scss:19]: add space

File: client/styles/pages/sputnikdaos.scss [hunk 6]
```diff
@@ -0,0 +1,22 @@
+@import 'client/styles/shared';
+
+.SputnikDAOsPage {
+    .title {
+        font-size: 32px;
+        margin-top: 32px;
+        padding-left: 12px;
+    }
+
+    .nearBadge{
```
Comment (zakhap) [inline client/styles/pages/sputnikdaos.scss:10]: add space

**Review by zakhap — CHANGES_REQUESTED** [review]
I'll help with the BN fixes tomorrow, but there are still a bunch of other little fixes that need to be made

File: client/scripts/views/pages/sputnikdaos.ts [hunk 7]
```diff
@@ -0,0 +1,111 @@
...
+  proposalPeriod: string;
+  bountyBond: string;
+  bountyPeriod: string;
+  council: string[];
+}
+
+const SputnikDAOsPage : m.Component<{}, { daosRequested: boolean, daosList: IDaoInfo[] }> = {
+  view: (vnode) => {
+    const activeEntity = app.community ? app.community : app.chain;
+    var allCommunities = (app.config.communities.getAll() as (CommunityInfo | ChainInfo)[]).concat(app.config.chains.getAll());
```
[the remark this answers is not in the document]
Comment (true-eye) [reply client/scripts/views/pages/sputnikdaos.ts:31]: "also, we should sort the communities first if they exist on cw, then by bond." - Zak Hap, 2021/11/03
It is for checking if the dao name exists on the communities names list. If it exists, it's sorted on top.

**Review by zakhap — APPROVED** [review]

**Pull request merged**